### PR TITLE
Create New Card GUI's

### PR DESCRIPTION
**Pull request by stuhops**
Tested slightly, but impossible because the latest build of actual-uno-deck is broken

### CODE REVIEW

**Comment by jak103** [comment]
@stuhops I'd like to get this merged to dev, can you make sure it is working against the latest dev and then we can merge it and close it?

**Comment by bjsvedin** [comment]
I have resolved conflicts by keeping everything on dev.

**Comment by jak103** [comment]
Looks like this isn't building. Fix the build issues and then we can get it merged. I'm really excited for this one.

**Comment by logan-hendricks** [comment]
This one might just have some merge conflicts due to later changes we just brought in. If someone wants to work on that tomorrow with me. I'd be glad to help resolve the latest changes and get this branch ready to merge in.

**Comment by bjsvedin** [comment]
I merged dev back into this one for you. Basically the only thing changed was cards.vue so i accepted all from dev when it came to conflicts. It should now build and pass all checks.

**Comment by logan-hendricks** [comment]
I'm noticing this branch is throwing a nil pointer error.

```
ine":"73","message":"[PANIC RECOVER] runtime error: invalid memory address or nil pointer dereference goroutine 14 [running]:\ngithub.com/labstack/echo/v4/middleware.RecoverWithConfig.func1.1.1(0xd859b8, 0x1000, 0x0, 0xf30340, 0xc000119040)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/recover.go:71 +0xee\npanic(0xc4bea0, 0x14b7f20)\n\t/usr/local/go/src/runtime/panic.go:969 +0x166\nmain.buildGameState(0x0, 0xc0000c8240, 0x18, 0x0)\n\t/server/routeHandlers.go:264 +0x45\nmain.getGameState(0xf30340, 0xc000119040, 0x4, 0xc4f6c0)\n\t/server/routeHandlers.go:157 +0x111\ngithub.com/labstack/echo/v4/middleware.JWTWithConfig.func2.1(0xf30340, 0xc000119040, 0xc0000620f0, 0xc0000620f0)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/jwt.go:207 +0x3c0\ngithub.com/labstack/echo/v4.(*Echo).add.func1(0xf30340, 0xc000119040, 0xf09f40, 0xc0000620f0)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/echo.go:512 +0x8a\ngithub.com/labstack/echo/v4/middleware.StaticWithConfig.func1.1(0xf30340, 0xc000119040, 0xd5e96d, 0x1b)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/static.go:169 +0x2b9\ngithub.com/labstack/echo/v4/middleware.CORSWithConfig.func1.1(0xf30340, 0xc000119040, 0x7f62795d8700, 0x0)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/cors.go:121 +0x477\ngithub.com/labstack/echo/v4/middleware.RecoverWithConfig.func1.1(0xf30340, 0xc000119040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/recover.go:78 +0x12a\ngithub.com/labstack/echo/v4/middleware.GzipWithConfig.func1.1(0xf30340, 0xc000119040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/middleware/compress.go:92 +0x20c\ngithub.com/labstack/echo/v4.(*Echo).ServeHTTP(0xc00000cb40, 0xf1b380, 0xc0003a82a0, 0xc000018300)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/echo.go:623 +0x16c\nnet/http.serverHandler.ServeHTTP(0xc0003a8000, 0xf1b380, 0xc0003a82a0, 0xc000018300)\n\t/usr/local/go/src/net/http/server.go:2807 +0xa3\nnet/http.(*conn).serve(0xc000118320, 0xf1d0c0, 0xc000076000)\n\t/usr/local/go/src/net/http/server.go:1895 +0x86c\ncreated by net/http.(*Server).Serve\n\t/usr/local/go/src/net/http/server.go:2933 +0x35c\n\ngoroutine 1 [IO wait]:\ninternal/poll.runtime_pollWait(0x7f6250107000, 0x72, 0x0)\n\t/usr/local/go/src/runtime/netpoll.go:203 +0x55\ninternal/poll.(*pollDesc).wait(0xc000152818, 0x72, 0x0, 0x0, 0xd4f865)\n\t/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45\ninternal/poll.(*pollDesc).waitRead(...)\n\t/usr/local/go/src/internal/poll/fd_poll_runtime.go:92\ninternal/poll.(*FD).Accept(0xc000152800, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)\n\t/usr/local/go/src/internal/poll/fd_unix.go:384 +0x1d4\nnet.(*netFD).accept(0xc000152800, 0x33, 0x14f9998, 0x203000)\n\t/usr/local/go/src/net/fd_unix.go:238 +0x42\nnet.(*TCPListener).accept(0xc00000fc40, 0x29e8d60804, 0x2, 0xef296a)\n\t/usr/local/go/src/net/tcpsock_posix.go:139 +0x32\nnet.(*TCPListener).AcceptTCP(0xc00000fc40, 0xe4b22c852a835882, 0x1000000000000, 0xe4b22c852a835882)\n\t/usr/local/go/src/net/tcpsock.go:248 +0x65\ngithub.com/labstack/echo/v4.tcpKeepAliveListener.Accept(0xc00000fc40, 0xc0000e9c20, 0x4bfe56, 0x5f27a286, 0x43f43e)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/echo.go:859 +0x2f\nnet/http.(*Server).Serve(0xc0003a8000, 0xf19600, 0xc000010538, 0x0, 0x0)\n\t/usr/local/go/src/net/http/server.go:2901 +0x25d\ngithub.com/labstack/echo/v4.(*Echo).StartServer(0xc00000cb40, 0xc0003a8000, 0x1, 0x1)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/echo.go:714 +0x385\ngithub.com/labstack/echo/v4.(*Echo).Start(...)\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.1.16/echo.go:634\nmain.main()\n\t/server/main.go:38 +0x56d\n\ngoroutine 6 [select]:\ngo.opencensus.io/stats/view.(*worker).start(0xc00007a730)\n\t/go/pkg/mod/go.opencensus.io@v0.22.3/stats/view/worker.go:154 +0x100\ncreated by go.opencensus.io/stats/view.init.0\n\t/go/pkg/mod/go.opencensus.io@v0.22.3/stats/view/worker.go:32 +0x57\n\ngoroutine 8 [IO wait]:\ninternal/poll.runtime_pollWait(0x7f6250106f20, 0x72, 0xffffffffffffffff)\n\t/usr/local/go/src/runtime/netpoll.go:203 +0x55\ninternal/poll.(*pollDesc).wait(0xc000152898, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)\n\t/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45\ninternal/poll.(*pollDesc)\n"}
```
Not sure why. I'm not seeing it on dev though.

File: client/src/components/Card.vue [hunk 1]
```diff
@@ -1,9 +1,111 @@
...
     <span class="inner">
       <span class="mark">{{ number }}</span>
     </span>
   </div>
+  <div v-else-if="number === 'skip'" :class="card_classes">
+    <span class="inner">
+      <span class="mark">S</span>
+    </span>
+  </div>
+  <div v-else-if="number === 'draw_two'" :class="card_classes">
```
Comment (logan-hendricks) [inline client/src/components/Card.vue:12]: This number was updated to D2

File: client/src/components/Card.vue [hunk 2]
```diff
@@ -1,9 +1,111 @@
 <template>
-  <div :class="card_specifics">
+  <div v-if="typeof number === 'number'" :class="card_classes">
     <span class="inner">
       <span class="mark">{{ number }}</span>
     </span>
   </div>
+  <div v-else-if="number === 'skip'" :class="card_classes">
```
Comment (logan-hendricks) [inline client/src/components/Card.vue:7]: This number was also updated to S

File: client/src/components/Card.vue [hunk 3]
```diff
@@ -1,9 +1,111 @@
...
+    <span class="inner">
+      <span class="mark">S</span>
+    </span>
+  </div>
+  <div v-else-if="number === 'draw_two'" :class="card_classes">
+    <span class="inner">
+      <span class="mark">D</span>
+    </span>
+  </div>
+  <div v-else-if="number === 'wild_draw_four'" :class="card_classes">
```
Comment (logan-hendricks) [inline client/src/components/Card.vue:17]: I think this was updated to W4

File: client/src/components/Card.vue [hunk 4]
```diff
@@ -1,9 +1,111 @@
...
+    <span class="inner">
+      <span class="mark">4</span>
+    </span>
+  </div>
+  <div v-else-if="number === 'wild'" :class="card_classes">
```
Comment (logan-hendricks) [inline client/src/components/Card.vue:22]: and this has been updated to W

File: client/src/components/Card.vue [hunk 5]
```diff
@@ -1,9 +1,111 @@
...
+    <span class="inner">
+      <span class="mark">w</span>
+    </span>
+  </div>
+  <div v-else-if="number === 'reverse'" :class="card_classes">
```
Comment (logan-hendricks) [inline client/src/components/Card.vue:27]: This was now set to R

**Comment by stuhops** [comment]
Okay, I've been working on it. When I wrote the code the dev branch was broken so I wasn't able to test as I wrote so it'll have a few bugs. I shouldn't need any help at this point though. Thanks for offering!

**Pull request merged**